src/utils/mobileOSDetectAsync.utils.ts

```diff
@@ -0,0 +1,93 @@
+import UAParser from "ua-parser-js";
+import { huaweiDevicesRegex, validCodes } from "../constants/mobile-devices.constants";
+
+/**
+ * This file contains utility functions and types for detecting the mobile operating system.
+ * It uses the User-Agent string and the User-Agent Client Hints API to determine the OS.
+ */
+
+declare global {
+    interface Window {
+        MSStream?: {
+            msClose: () => void;
+            msDetachStream: () => void;
+            readonly type: string;
+        };
+        opera?: string;
```

Comment:
Add some comments on why these typedefs are added. Will these globals only exist in the respective browser vendors? Or link to some ref so that people know what this is. Add globals only when necessary as it may pollute the global types. Only extend if it is absolutely necessary.

Please read this before deciding: https://www.totaltypescript.com/how-to-properly-type-window